substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -204,26 +226,36 @@ fn call_host_function(
 	{
 		Ok(value) => value,
 		Err(error) => {
-			log::warn!("Call into the host function '{}' failed: {error}", function.name());
-			return Err(polkavm::Trap::default());
+			return Err(polkavm::CallError::User(format!(
+				"Call into the host function '{}' failed: {error}",
```

Comment:
Since we're appending/chaining the error messages make this start with lowercase

(e.g. we're doing "error foo: error bar: error baz")